rules_daml/daml-build.bzl

```
@@ -0,0 +1,130 @@
+# Copyright (c) 2020 The DAML Authors. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+def file_of_target(k):
+    [file] = k.files.to_list()
+    return file
+
+_damlc = attr.label(
+    allow_single_file = True,
+    default = Label("//compiler/damlc"),
+    executable = True,
+    cfg = "host",
+)
+
+def _daml_build_impl(ctx):
+    name = ctx.label.name
+    config = ctx.file.config
+    damls = ctx.files.damls
+    dd = ctx.attr.dar_dict
+    damlc = ctx.file._damlc
+    input_dars = [file_of_target(k) for k in dd.keys()]
+    output_dar = ctx.actions.declare_file(name + ".dar")
+    ctx.actions.run_shell(
+        tools = depset([damlc]),
+        inputs = depset([config] + damls + input_dars),
+        outputs = [output_dar],
+        progress_message = "Building DAML project %s" % name,
+        command = """
+            set -eou pipefail
+            tmpdir=$(mktemp -d)
+            mkdir -p $tmpdir/daml
```

```suggestion
            trap "rm -rf $tmpdir" EXIT
            mkdir -p $tmpdir/daml
```
Leaving stale directories around is a bit uggly and might result in issues in the long term (although somewhat unlikely at least on CI given that we recycle nodes but it’s easy enough to clean it up).